Sdk/ArgumentFormatter.cs

```diff
@@ -166,6 +166,12 @@ static string Format(object value, int depth, out int? pointerPostion, int? erro
 					// In these cases, just continue formatting.
 				}
 
+				if (value is float)
+					return $"{value:G7}";
+
+				if (value is double)
+					return $"{value:G19}";
```

Comment:
This should be `G17` to guarantee uniqueness and so is overly precise and may result in longer strings than necessary.

In both cases, `R` might be a suitable alternative. It will produce the shortest roundtrippable string on .NET Core 3.0+ and will produce (barring a couple known bugs) the shorter of 15 or 17 digit strings on .NET Framework.